x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDenseVector.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.convert;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.compute.ann.ConvertEvaluator;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.esql.core.type.DataType.DENSE_VECTOR;
+import static org.elasticsearch.xpack.esql.core.type.DataType.DOUBLE;
+import static org.elasticsearch.xpack.esql.core.type.DataType.INTEGER;
+import static org.elasticsearch.xpack.esql.core.type.DataType.KEYWORD;
+import static org.elasticsearch.xpack.esql.core.type.DataType.LONG;
+
+/**
+ * Converts a multi-valued input of numbers, or a hexadecimal string, to a dense_vector.
+ */
+public class ToDenseVector extends AbstractConvertFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        Expression.class,
+        "ToDenseVector",
+        ToDenseVector::new
+    );
+
+    private static final Map<DataType, BuildFactory> EVALUATORS = Map.ofEntries(
+        Map.entry(DENSE_VECTOR, (source, fieldEval) -> fieldEval),
+        Map.entry(LONG, ToDenseVectorFromLongEvaluator.Factory::new),
+        Map.entry(INTEGER, ToDenseVectorFromIntEvaluator.Factory::new),
+        Map.entry(DOUBLE, ToDenseVectorFromDoubleEvaluator.Factory::new),
+        Map.entry(KEYWORD, ToDenseVectorFromStringEvaluator.Factory::new)
+    );
+
+    @FunctionInfo(
+        returnType = "dense_vector",
+        description = "Converts a multi-valued input of numbers, or a hexadecimal string, to a dense_vector.",
+        examples = @Example(file = "dense_vector", tag = "to_dense_vector-ints")
+    )
+    public ToDenseVector(
+        Source source,
+        @Param(
+            name = "field",
+            type = { "double", "long", "integer", "keyword" },
```

Comment:
Should we add text as well (if it is loaded from an ES text field).

Reply:
I would say no - it's ok to allow for hex string constants, but I don't think they should be analyzed in any way or retrieved from an analyzed field. 

If any time this is needed, they can be converted via the `::keyword` casting operator.